twilio/rest/__init__.py

```diff
@@ -39,28 +29,7 @@ def __init__(self, username=None, password=None, account_sid=None, region=None,
         :returns: Twilio Client
         :rtype: twilio.rest.Client
         """
-        environment = environment or os.environ
-
-        self.username = username or environment.get('TWILIO_ACCOUNT_SID')
-        """ :type : str """
-        self.password = password or environment.get('TWILIO_AUTH_TOKEN')
-        """ :type : str """
-        self.account_sid = account_sid or self.username
-        """ :type : str """
-        self.edge = edge or environment.get('TWILIO_EDGE')
-        """ :type : str """
-        self.region = region or environment.get('TWILIO_REGION')
-        """ :type : str """
-        self.user_agent_extensions = user_agent_extensions or []
-        """ :type : list[str] """
-
-        if not self.username or not self.password:
-            raise TwilioException("Credentials are required to create a TwilioClient")
-
-        self.auth = (self.username, self.password)
-        """ :type : tuple(str, str) """
-        self.http_client = http_client or TwilioHttpClient()
-        """ :type : HttpClient """
+        super().__init__(username, password, account_sid, region, http_client, environment, edge, user_agent_extensions)
 
         # Domains
         self._accounts = None
```

Comment:
I think the constructor in this class can just be removed since setting these to `None` doesn't really have an effect.

Reply:
Are you suggesting that I removed the #Domains too? cause that will cause issues in accessing it (test cases will look for `account` instead of `_account`). I could shift the #Domains to class attributes (`._accounts=None`)

Reply:
Yes, that's what I was thinking. The underscore attributes are just for lazy-loading/caching. Are the tests dependent on them?

Reply:
I think issue comes from the `@property` accessors. If I remove the underscore attributes, it will attempt to find it - `'Client' object has no attribute '_conversations'. Did you mean: 'conversations'?` I think like we have to declare the attributes

Reply:
Ah, looks like python requires instance variables to be defined in the constructor. Never mind.